Gems/ROS2/Code/Source/ROS2SystemComponent.cpp

```diff
@@ -138,6 +139,10 @@ namespace ROS2
         m_loadTemplatesHandler.Disconnect();
         m_dynamicTFBroadcaster.reset();
         m_staticTFBroadcaster.reset();
+        m_executor->remove_node(m_ros2Node);
+        m_executor.reset();
+        m_simulationClock->ResetPublisher();
+        m_ros2Node.reset();
```

Comment:
I can see reasons why the node should be up, such as ROS parameters. However, the change can be easily introduced with such feature.

Reply:
I'm not quite sure what you are suggesting. Are you implying that the node should be active outside of the simulation and that it should be activated immediately after resetting?

Reply:
Yes, a node which is active outside of simulation mode could support setting of ROS [parameters](https://docs.ros.org/en/humble/Tutorials/Beginner-CLI-Tools/Understanding-ROS2-Parameters/Understanding-ROS2-Parameters.html#ros2-param-set) on actual Editor Components, which is quite a valid use-case.

However, this could be changed once we implement parameters.